### PR TITLE
feat: add translations

### PR DESCRIPTION
**Pull request by ReidyT**

### CODE REVIEW

File: src/config/i18n.ts [hunk 1]
```diff
@@ -1,28 +1,32 @@
...
-  },
-});
+export const useTextAnalysisTranslation = () =>
+  useTranslation(TEXT_ANALYSIS_NAMESPACE);
+export const useCommonTranslation = () => useTranslation(namespaces.common);
+export const useMessagesTranslation = () => useTranslation(namespaces.messages);
+export const useEnumsTranslation = () => useTranslation(namespaces.enums);
+export const useCategoriesTranslation = () =>
+  useTranslation(namespaces.categories);
+export const useChatboxTranslation = () => useTranslation(namespaces.chatbox);
```
Comment (spaenleh) [inline src/config/i18n.ts:30]: I don't think you need all these namespaces.

File: src/config/i18n.ts [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
 
+import buildI18n, { namespaces } from '@graasp/translations';
+
+import ar from '../langs/ar.json';
+import de from '../langs/de.json';
 import en from '../langs/en.json';
 import fr from '../langs/fr.json';
+import it from '../langs/it.json';
+
+const i18n = buildI18n().use(initReactI18next);
```
Comment (spaenleh) [inline src/config/i18n.ts:14]: This function allows to define the default namespace to use. Set it to be the Text analysis namespace.

File: src/components/common/settings/SetText.tsx [hunk 3]
```diff
@@ -31,15 +31,14 @@ const SetText: FC<Prop> = ({
...
-      sx={{ margin: DEFAULT_MARGIN }}
     >
       <TextField
         data-cy={textDataCy}
         multiline={multiline}
         label={textFieldLabel}
         variant="outlined"
         onChange={handleChange}
-        sx={{ width: FULL_WIDTH, marginRight: DEFAULT_MARGIN }}
+        sx={{ width: FULL_WIDTH }}
```
Comment (spaenleh) [inline src/components/common/settings/SetText.tsx:41]: There is a prop `fullWidth` on the TextField component. [Ref](https://mui.com/material-ui/react-text-field/#full-width)

File: src/components/common/settings/KeyWords.tsx [hunk 4]
```diff
@@ -152,19 +152,20 @@ const KeyWords: FC<Prop> = ({
...
         flexDirection={{ xs: 'column', sm: 'row' }}
-        rowGap={2}
+        rowGap={DEFAULT_IN_SECTION_SPACING}
+        columnGap={2}
       >
         <TextField
           data-cy={ENTER_KEYWORD_FIELD_CY}
           label="Enter the keyword"
-          sx={{ width: FULL_WIDTH, marginRight: DEFAULT_MARGIN }}
+          sx={{ width: FULL_WIDTH }}
```
Comment (spaenleh) [inline src/components/common/settings/KeyWords.tsx:161]: Same here

File: src/components/common/settings/KeyWords.tsx [hunk 5]
```diff
@@ -152,19 +152,20 @@ const KeyWords: FC<Prop> = ({
...
           value={keywordDef.word}
           onChange={(e) => updateKeywordDefinition('word', e.target)}
         />
         <TextField
           data-cy={ENTER_DEFINITION_FIELD_CY}
           label="Enter the keyword's definition"
-          sx={{ width: FULL_WIDTH, marginRight: DEFAULT_MARGIN }}
+          sx={{ width: FULL_WIDTH }}
```
Comment (spaenleh) [inline src/components/common/settings/KeyWords.tsx:168]: And here

File: src/config/i18n.ts
Comment (spaenleh) [inline src/config/i18n.ts:1]: Not dure if you need all the translations from @graasp/translations ??

**Review by spaenleh — CHANGES_REQUESTED** [review]
Nice lets talk about the translations IRL

File: src/config/i18n.ts (hunk 2 above)
Comment (ReidyT) [reply src/config/i18n.ts:14]: Okay I will redo

**Review by spaenleh — APPROVED** [review]

**Pull request merged**